### ISSUE TITLE
sfdx force:source:deploy command fails with ERROR running force:source:deploy: Metadata API request failed: Unexpected internal servlet state whereas deployment is actually successful on org.

### ISSUE THREAD

**Issue by FabienTaillon** [forcedotcom/cli#1811]
### Summary

`sfdx force:source:deploy` command fails with `ERROR running force:source:deploy:  Metadata API request failed: Unexpected internal servlet state` whereas deployment is actually successful on org.

### Steps To Reproduce:

Unfortunately this happens randomly, but in the past week it happened to me very frequently. With the exact same org and package it works when I try again. As the deployment is successful on the org it should be something with the CLI.

As you can see below, Deployment Id is the same on the CLI and on the org but the CLI fails with an error:

<img width="785" alt="1" src="https://user-images.githubusercontent.com/2465122/202392213-8e0e07c2-6b6d-4c8f-bf98-c2a23d737c2c.png">

<img width="1372" alt="2" src="https://user-images.githubusercontent.com/2465122/202392237-c84970b5-e094-4126-bfc7-aad8296458ea.png">

In this exemple it was a validation, but it occurs whereas it's a validation or a real deployment.

### Expected result

Deployment is successful on the CLI side.

### Actual result

CLI reports the following error:  `ERROR running force:source:deploy:  Metadata API request failed: Unexpected internal servlet state`.

### System Information

This is happening in our CI jobs. 
We're using the `salesforce/salesforcedx:latest-full` docker image.
CLI command is the following: `sfdx force:source:deploy --sourcepath force-app --testlevel RunLocalTests --targetusername CI --wait 30 --checkonly`

**Comment by mshanemc** [comment]
the code for what types of things we'll retry lives here: https://github.com/forcedotcom/sfdx-core/blob/0d2fa59837fd19ff07badb820b8764b78b86f7e5/src/status/pollingClient.ts#L68

**Comment by mshanemc** [comment]
There's code in SDR https://github.com/forcedotcom/source-deploy-retrieve/blob/58152775b14ae80c483ec83a1b80c4d737ecb351/src/client/metadataTransfer.ts#L219

for handling intermittent problems with the metadata API.  If you want to add your error there, you could try it with the next docker image to see if it makes a difference.

That's a hard one to test--looks like you have the error message that you could try matching by?

It's also possible by the time we get this in, whatever the server has going on has been fixed.

**Comment by FabienTaillon** [comment]
Thanks. In the meantime I've found a Known Issue for it, even though it doesn't mention the Metadata API in it:  https://trailblazer.salesforce.com/issues_view?id=a1p4V000002e0HyQAI

If the issue doesn't get resolved I'll do a PR to add this error message.

**Comment by matt-newell** [comment]
I'm seeing similar issue where a deploy to a sandbox succeeded but the deploy reports UNKNOWN_EXCEPTION: 

The command I'm running `sfdx force:mdapi:deploy -d deploy -u OrgAliase -w 40 -l NoTestRun --checkonly`

```ERROR running force:mdapi:deploy:  Metadata API request failed: UNKNOWN_EXCEPTION: java.io.IOException Unable to create phoenix connection. DualCluster:true; Error: All futures failed. HAGroup:HighAvailabilityGroup{roleRecord=ClusterRoleRecord{haGroupName='acsopdrHA-hbase3a-ia4', policy=PARALLEL, zk1='hbase3a-mnds1-1-ia4.ops.sfdc.net,hbase3a-mnds2-1-ia4.ops.sfdc.net,hbase3a-mnds3-1-ia4.ops.sfdc.net,hbase3a-mnds4-1-ia4.ops.sfdc.net,hbase3a-mnds5-1-ia4.ops.sfdc.net:2181:/hbase', role1=ACTIVE, zk2='hbase3a-mnds1-1-ia5.ops.sfdc.net,hbase3a-mnds2-1-ia5.ops.sfdc.net,hbase3a-mnds3-1-ia5.ops.sfdc.net,hbase3a-mnds4-1-ia5.ops.sfdc.net,hbase3a-mnds5-1-ia5.ops.sfdc.net:2181:/hbase', role2=STANDBY, version=1}, state=READY}``` 

<img width="341" alt="Screen Shot 2022-11-18 at 12 15 24 PM" src="https://user-images.githubusercontent.com/7570953/202763190-e80ef558-fb6e-42ee-9638-d7c24c9770e0.png">

But a few minutes after the CI/CD job fails the deploy the org Deployment Status shows a grey Vaidate Succeeded

**Comment by renanlq** [comment]
From our side, we are facing a similar scenario: deploy status error with deployment succeded in our Sandbox, here is some part of our CI Log:
```
15/05/2023 12:29:35  + sfdx --version
15/05/2023 12:29:37  sfdx-cli/7.186.2 linux-x64 node-v18.13.0
...
15/05/2023 12:30:31  + sfdx force:source:deploy -p ./build/force-app/main/default -u **** -l RunLocalTests -w 240 --verbose
15/05/2023 12:30:32   ›   Warning: sfdx-cli update available from 7.186.2 to 7.200.7.
15/05/2023 12:30:34  Deploying v56.0 metadata to **** using the v58.0 SOAP API
15/05/2023 12:30:35  Deploy ID: 0Af********0A2
15/05/2023 12:30:39  DEPLOY PROGRESS | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0/3 Components
15/05/2023 12:30:41  DEPLOY PROGRESS | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0/3 Components
15/05/2023 12:30:43  DEPLOY PROGRESS | █████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 1/3 Components
15/05/2023 12:30:45  DEPLOY PROGRESS | █████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 1/3 Components
15/05/2023 12:30:47  DEPLOY PROGRESS | ████████████████████████████████████████ | 3/3 Components
...
15/05/2023 12:49:32  DEPLOY PROGRESS | █████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 967/3060 Components
15/05/2023 12:49:33  (node:2310) Warning: Network error occurred.  Continuing to poll.
15/05/2023 12:49:33  (Use `node --trace-warnings ...` to show where the warning was created)
15/05/2023 12:49:34  DEPLOY PROGRESS | █████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 967/3060 Components
...
15/05/2023 13:01:48  DEPLOY PROGRESS | ████████████████████░░░░░░░░░░░░░░░░░░░░ | 1549/3060 Components
15/05/2023 13:01:50  (node:2310) Warning: Network error occurred.  Continuing to poll.
15/05/2023 13:01:50  DEPLOY PROGRESS | ████████████████████░░░░░░░░░░░░░░░░░░░░ | 1550/3060 Components
...
15/05/2023 13:06:22  DEPLOY PROGRESS | ████████████████████████░░░░░░░░░░░░░░░░ | 1830/3060 Components
15/05/2023 13:06:23  (node:2310) Warning: Network error occurred.  Continuing to poll.
15/05/2023 13:06:24  DEPLOY PROGRESS | ████████████████████████░░░░░░░░░░░░░░░░ | 1840/3060 Components
...
15/05/2023 13:28:34  DEPLOY PROGRESS | █████████████████████████████████████░░░ | 2806/3060 Components
15/05/2023 13:28:34  
15/05/2023 13:28:34  ERROR running force:source:deploy:  Metadata API request failed: FUNCTIONALITY_TEMPORARILY_UNAVAILABLE: Failed while fetching query cursor data for this QueryLocator (org 00D*******jT6, path 0r8x*******q7G), please retry your request.
```
We are in progress to update our version, due to the fact that our security team are validating latest versions from SFDX CLI.

Additionally, are evaluating the possibility of add retry pattern from our side, to mitigate this scenario. Could you give us more info, that would be important when develop retry to keep our CI job running without network connection error with Salesforce environment?

**Comment by mshanemc** [comment]
also handled by https://github.com/forcedotcom/source-deploy-retrieve/blob/64132ca9a3353dda8810658bd077e495923cfba5/src/client/metadataTransfer.ts#L233
